feat: remove applied coupon code

stravo1 opened the pull request:
fixes #290

stravo1 committed:
feat: show & remove applied coupon code

NagariaHussain commented:
Add screenshots / video or GIF for UI changes.

NagariaHussain commented on webshop/templates/includes/cart/coupon_code.html:
```diff
@@ -0,0 +1,32 @@
+</div>
+<div class="d-flex align-items-center text-muted my-1">
+    <hr class="flex-grow-1" />
+    <span class="px-2">OR</span>
```
As a general convention, "transformations" should be through CSS. 

```suggestion
    <span class="px-2 text-uppercase">or</span>
```

stravo1 replied:
fixed at 6a83c1e3b7180739848f5eba19c52076c74c15aa

NagariaHussain commented on webshop/webshop/shopping_cart/cart.py:
```diff
@@ -799,3 +800,20 @@ def apply_coupon_code(applied_code, applied_referral_sales_partner):
+	if quotation.has_value_changed("coupon_code") and not quotation.coupon_code:
+		quotation.discount_amount = 0
+		quotation.ignore_pricing_rule = 1
+		quotation.save()
```
Save should happen once at the end? has_value_changed only makes sense for unsaved doc.

stravo1 replied:
fixed at a60708b6fb2a7c14865f2b3fe423a9bf0a048c5c

stravo1 committed:
chore(ui): update "OR" text to lowercase and apply text-uppercase class for consistency

stravo1 commented:
> Add screenshots / video or GIF for UI changes.

<img width="1440" alt="Screenshot 2025-04-17 at 12 13 17 AM" src="https://github.com/user-attachments/assets/deebc0ba-32b0-4e99-8bc9-d9637f522c63" />

https://github.com/user-attachments/assets/fea425dc-6835-4b4e-803b-850b87e656d3

stravo1 committed:
fix: saving doc only once while removing coupon code

- had the misconception that has_value_changed is accessible only after a save
- resetting additional_discount_percentage as it gets calculated while processing the pricing rule but we are ignoring it here while saving

NagariaHussain commented:
* Let's make the coupon code a secondary button ([blog](https://buildwithhussain.com/articles/five-ui-ux-things))
* Instead of showing alternative coupon code UI, we can show it back once they remove

NagariaHussain commented on webshop/webshop/shopping_cart/cart.py:
```diff
@@ -799,3 +800,22 @@ def apply_coupon_code(applied_code, applied_referral_sales_partner):
+
+	# reset discount amount if coupon code is removed (on desk it is done in client side)
+	# as we are enabling ignore_pricing_rule, so we also need to manually reset discount percentage
+	if quotation.has_value_changed("coupon_code") and not quotation.coupon_code:
```
This will always evaluate to true.

stravo1 replied:
fixed at 8274f2fcebb3c6aeb8b387ac7ab84a74164b7352

stravo1 committed:
fix(ui): made btn-secondary and improved coupon code ui

stravo1 committed:
fix: removed `if` as it always evaluates to true

stravo1 commented:
> * Let's make the coupon code a secondary button ([blog](https://buildwithhussain.com/articles/five-ui-ux-things))
> * Instead of showing alternative coupon code UI, we can show it back once they remove

fiexd at ac786e5755f68901f7324ad6843351560bf0481d

stravo1 commented:
Screenshots after UI review:


https://github.com/user-attachments/assets/431ec8ef-dcdd-405c-b288-58b5dadd11fb

<img width="1440" alt="Screenshot 2025-04-22 at 10 33 01 PM" src="https://github.com/user-attachments/assets/5f82e07f-f1a3-4b02-946d-895e2d813f46" />
<img width="1440" alt="Screenshot 2025-04-22 at 10 56 12 PM" src="https://github.com/user-attachments/assets/ff04a132-a256-4e85-b86e-44b6c01bb2e6" />

NagariaHussain commented on webshop/templates/includes/cart/coupon_code.html:
```diff
@@ -0,0 +1,28 @@
+    class="cart-coupon-code flex gap-2 align-items-center w-fit rounded mb-4"
+    style="border: 2px dashed #ccc;"
+>
+    <div class="p-2 small"> {{ coupon_code}} </div>
```
```suggestion
    <div class="px-2 py-1 small"> {{ coupon_code}} </div>
```

NagariaHussain committed:
fix: lesser y padding

NagariaHussain commented on webshop/webshop/shopping_cart/cart.py:
```diff
@@ -799,3 +800,21 @@ def apply_coupon_code(applied_code, applied_referral_sales_partner):
 	return quotation
+
+ 
+@frappe.whitelist(allow_guest=True)
```
Why is allow_guest required here @stravo1 ?

NagariaHussain replied:
Got it, apply also has it, I presume it will be for guest users.

NagariaHussain commented:
Congratulations on your first PR merge on webshop @stravo1  🎉

stravo1 commented:
> Congratulations on your first PR merge on webshop @stravo1  🎉

thanks ✨
